spark/src/main/scala/org/apache/spark/sql/delta/TypeWideningMode.scala

```diff
@@ -16,44 +16,96 @@
 
 package org.apache.spark.sql.delta
 
-import org.apache.spark.sql.types.AtomicType
+import org.apache.spark.sql.util.ScalaExtensions._
+
+import org.apache.spark.sql.catalyst.analysis.DecimalPrecisionTypeCoercion
+import org.apache.spark.sql.types.{AtomicType, DecimalType}
 
 /**
  * A type widening mode captures a specific set of type changes that are allowed to be applied.
  * Currently:
  *  - NoTypeWidening: No type change is allowed.
- *  - AllTypeWidening: All supported type widening changes are allowed.
- *  - TypeEvolution(uniformIcebergCompatibleOnly = true): Type changes that are eligible to be
```

Comment:
I think we should still mention in here that some of the have the Iceberg compat flag

Reply:
I added back a small description of the iceberg compat flag